### ISSUE TITLE
MessageBag return full array

### ISSUE THREAD

**Issue by hondnl** [alexgarrett/violin#84]
First of all thank you for your very nice validation script, nicely done!

Did I miss something or is there no method to receive the whole error array back ? ($messages)
The all() method returns all the values..
What if I want the full array  ( to mix and match auto  on a form for example) ?
Would be nice to include it as default ( ->full() )  , now I have to write an extension for it or have to touch your original script.

thanks!

**Comment by tomchamp** [comment]
So to answer above, I just did the following, for me it works great.
``` 
$err = $vi->errors()->all();
$key = $vi->errors()->keys();

for($a=0; $a<sizeof($err); $a++) {
   for($b=0; $b<sizeof($key); $b++) {
      $errs[$key[$b]] = $err[$b];
   }
}
```

**Comment by pedzed** [comment]
@tomchamp Yeah, but it's not really nice, is it? A built-in solution would be better.

On another note, please never use `sizeof()`. http://stackoverflow.com/a/3974424

> I would use count() if they are the same, as in my experience it is more common, and therefore will cause less developers reading your code to say "sizeof(), what is that?" and having to consult the documentation.

In my whole PHP career I have never heard of `sizeof()`. I had to google that one.

**Comment by tomchamp** [comment]
No it's not the ideal solution, but it's a temporary one at best, and works perfectly fine. 

Aside from readability for developers they're hardly much different and personally I think they're both acceptable for small cases like this.

**Comment by pedzed** [comment]
> I think they're both acceptable for small cases like this.

Perhaps you misunderstood me. I said that I haven't seen `sizeof()` in PHP. `count()` is used so much more often. Just use `count()` instead of `sizeof()`?

**Comment by tomchamp** [comment]
I think you may have also misunderstood me, i'm simply saying their practically identical(Alias'), and its down to the developer which once they choose to use. In my eyes they're both pretty clear what they do, i highly doubt readability is an issue here, having to google what `sizeof()` does is really a bit far fetched in this instance. I'll quite happily use both as performance really isn't a factor between these functions. 

Aside from bickering about whether to use `count()` or` sizeof()`, maybe we should be looking to progress the current outstanding issue within Violin, which is why I posted the above snippet of code, granted a bit of a hacky solution.

**Comment by pedzed** [comment]
> In my eyes they're both pretty clear what they do, i highly doubt readability is an issue here, having to google what sizeof() does is really a bit far fetched in this instance.

Perhaps you haven't programmed in other languages, but for me, it's definitely not clear what it does. In PHP, yes, it is an alias. In other programming languages it is not, which is why it confused me in the first place.

> In the programming languages C and C++, the unary operator sizeof generates the size of a variable or datatype, measured in the number of char size storage units required for the type.
>
> [...]
>
> The sizeof operator is used to determine the amount of space a designated datatype would occupy in memory.
>
> _From: https://en.wikipedia.org/wiki/Sizeof_

And many other people agree with me.

![image](https://cloud.githubusercontent.com/assets/8146593/15288981/4b4b0bdc-1b6f-11e6-85a0-381d73176847.png)

> we should be looking to progress the current outstanding issue within Violin, which is why I posted the above snippet of code, granted a bit of a hacky solution.

Yeah, but there are already quite a few PR's open. I don't think it would be accepted any soon, but you can try if you want.

**Comment by tomchamp** [comment]
Thank you pedzed. Although i didn't need convincing, perfectly happy to use `count()` over sizeof(), quite honestly i don't much care.


> Yeah, but there are already quite a few PR's open. I don't think it would be accepted any soon, but you can try if you want.

In that case, i'll stick with my current code. It works, and I only wanted to share what with the community looking to help others.

**Comment by pedzed** [comment]
> i don't much care.

Sorry about all the fuss. It's just easier if things don't go against [POLS](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

> I only wanted to share what with the community.

Thanks for that.
